### PR TITLE
ARROW-17092: [Docs] Add note about "Feather" to the IPC file format document

### PR DESCRIPTION
**Pull request by eitsupi**

### CODE REVIEW

File: docs/source/format/Columnar.rst
Comment (lidavidm) [inline docs/source/format/Columnar.rst:1049]: Should we note that sometimes they still use the ".feather" extension?

**Review by lidavidm — APPROVED** [review]

File: docs/source/format/Columnar.rst
Comment (eitsupi) [reply docs/source/format/Columnar.rst]: I added a note about the `.feather` extension in ac36ed3c3a021d38b2eb67511ed2ce490fb54368.
I think either is fine and I will defer to the decision of the maintainers.

**Review by lidavidm — APPROVED** [review]

**Pull request merged**
